compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIECreateAIEWorkgroup.cpp

```diff
@@ -70,24 +70,58 @@ LogicalResult workgroupBuildForDmaCpyNdOp(
     Block::iterator controlCodeBegin, Block::iterator controlCodeEnd) {
   LLVM_DEBUG(llvm::dbgs() << "workgroupBuild [amdaie.dma_cpy_nd] Start\n");
   Attribute sourceMemSpace = dmaOp.getSourceObjectFifo().getMemorySpace();
+  Attribute targetMemSpace = dmaOp.getTargetObjectFifo().getMemorySpace();
```

Comment:
It would be useful to add a check that either `sourceMemSpace` or `targetMemSpace` is located on L3 and return an error if not. (I should have added that in the initial PR, just thinking of it now)